core/pouch/index.js

```diff
@@ -415,6 +436,41 @@ class Pouch {
     await this.createDesignDoc('byPath', query)
   }
 
+  // Create a view to find records based on their `local.path` attribute via a
+  // composite key.
+  // The key is separated in two parts:
+  // - the parent path
+  // - the document's name
+  // This allows us to either fetch documents via their full path or their
+  // parent path, recursively or not.
+  // The parent path of a document in the root folder will be '', not '.' as
+  // with Node's path.dirname() result.
+  async addByLocalPathView() {
+    const platform = JSON.stringify(process.platform)
+    const sep = JSON.stringify(path.sep)
+    const query = `function(doc) {
```

Comment:
I would have put the normalization per platform outside of the templated string:

```js
let normalized
switch (process.platform) {
case 'darwin':
    normalized = "doc.local.path.normalize('NFD').toUpperCase()"
    break;
case 'win32':
    normalized = "doc.local.path.toUpperCase()"
    break;
default:
    normalized = "doc.local.path"
}
const query = `function(doc) {
if ('local' in doc && 'path' in doc.local) {
    let normalized = ${normalized}
    // ...
}`
```

Reply:
It could work. I'll try it out.